ccm_web/server/src/auth/double.csrf.middleware.ts

```diff
@@ -0,0 +1,24 @@
+import { Injectable, NestMiddleware } from '@nestjs/common'
+import { doubleCsrf } from 'csrf-csrf'
+import { NextFunction, Request, Response } from 'express'
+import { Config } from '../config'
+import { ConfigService } from '@nestjs/config'
+
+import { AuthService } from './auth.service'
+
+@Injectable()
+export class DoubleCSRFProtectionMiddleware implements NestMiddleware {
+
+  constructor(
+    private readonly configService: ConfigService<Config, true>, 
+    private readonly authService: AuthService) {}
+  
+  use(req: Request, res: Response, next: NextFunction) {
+    const { doubleCsrfProtection } = doubleCsrf({
+      getSecret: () => this.configService.get<string>('server.csrfSecret', { infer: true }),
+      cookieName: '_Secure-ccm.x-csrf-token',
```

Comment:
I added a Cookie name with _Secure instead of _Host as per Authors recommendation. See the PR description for reasoning